custom_components/mc66c/sensor.py

```diff
@@ -92,7 +92,11 @@ def validateAndSetData(self, reading):
             _LOGGER.info("Successfully fetched new data: %s", reading)
 
             try:
-                data.append(int((reading[0]).decode("utf-8")) / 1000) # Energy.
+                for c in reading:
+                    if (len(c)) != 7:
+  	                    raise Exception('Received invalid datafield length')
```

Comment:
Looks like to much indentation 
```suggestion
                for c in reading:
                    if (len(c)) != 7:
  	                raise Exception('Received invalid datafield length')
```